### PR TITLE
Switch result panel and open website

### PR DESCRIPTION
**Pull request by Gelixxxx**
Finished 2 use case, Switch result panel and Open Website.
Test for each of their controller and interactor is also done and passes with 100% line coverage.
Plz correct me if bugs are noticed.
Screenshot 2023-12-04 211710

### CODE REVIEW

**Review by DonuttDog — COMMENTED** [review]
Great work on implementing the new features! However, I've observed that there are quite a few changes across multiple files. It seems like you might be working on an outdated version of the codebase.
For instance, in the Master branch, the SearchView.propertyChange() method includes additional cases from other contributors' work, but these changes are missing in your implementation. I'm concerned that your version might inadvertently overwrite these additions. This discrepancy appears in several files within the app, too. 
Therefore, in the merge process, **there might be conflict between them (or not, since you only miss code** rather than modifying them)

Coding style is pretty good, like name is followed convention, the structure is consistent with others.

The core logic is nice, and my concerns are all listed above. The program can successfully run.

Test are good, in my local IDE everything is fine, and the line coverage of controller and interactor all reach 100%.

All followed by CA engine, every layer is isolated to each and interface being a bridge here.

Maybe add some Javadoc, especially in the interactor, is a good idea.

The info is sufficient, and commits are easy to read.

**Comment by Gelixxxx** [comment]
Screenshot 2023-12-05 180857

File: src/app/Main.java [hunk 1]
```diff
@@ -102,21 +101,16 @@ public static void main(String[] args) {
 
         new ViewManager(views, cardLayout, viewManagerModel); // Finished adding views
 
-
         Map<String, JFrame> frames = new HashMap<>();
         frames.put(MainFrameViewModel.VIEW_NAME, mainFrame);
-
```
Comment (MarksonChen) [inline src/app/Main.java:108]: Why was my code deleted here?

File: src/app/Main.java [hunk 2]
```diff
@@ -102,21 +101,16 @@ public static void main(String[] args) {
...
-        HistoryView historyView = new HistoryView(historyViewModel, switchViewController, reuseHistoryQueryController, clearHistoryController);
-        frames.put(HistoryViewModel.VIEW_NAME, historyView);
-
-        StarredView starredView = new StarredView(searchViewModel, starredViewModel, fillDetailController, starController, openWebsiteController);
-        frames.put(StarredViewModel.VIEW_NAME, starredView);
-
```
Comment (MarksonChen) [inline src/app/Main.java:114]: Why are my codes deleted here?

File: src/data_access/WebDataAccessObject.java [hunk 3]
```diff
@@ -39,12 +31,6 @@ public String getResponse(String path) throws IOException {
         return response.body().string();
     }
 
-    /**
-     * It opens the web site in brower
-     * @param path is the url
-     * @throws URISyntaxException
-     * @throws IOException
-     */
```
Comment (MarksonChen) [inline src/data_access/WebDataAccessObject.java:47]: Why are Dennis's comments also deleted?

File: src/view/FrontPageView.java [hunk 4]
```diff
@@ -16,11 +16,11 @@ public FrontPageView(FrontPageViewModel frontPageViewModel, SwitchViewController
...
 
+        appTitle.addActionListener(evt ->
+                openWebsiteController.execute(FrontPageViewModel.TITLE_BUTTON_REDIRECT_URL)
+        );
 
-
-        searchButton.addActionListener(evt -> {
-            switchViewController.execute(SearchViewModel.VIEW_NAME);
-        });
+        // TODO add listeners
```
Comment (MarksonChen) [inline src/view/FrontPageView.java:23]: My codes are deleted again. There is definitely something wrong with how you handle merge conflicts.

**Comment by MarksonChen** [comment]
You seemed to have deleted other's pull requests by mistake when you merged pull requests. 
This pull request should not be accepted before you make sure you are not deleting any single line of other people's codes. 

I suggest that you: 
1. Open a new branch
2. Copy all your use case codes to that branch
3. Submit that branch as a pull request and delete this branch

**Pull request closed without merging**